### PR TITLE
test: unit tests for aws client

### PR DESCRIPTION
**Pull request by pawelpaszki**
## Description
test: unit tests for aws client

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- ~~[ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- ~~[ ] Required metrics/dashboards/alerts have been added (or PR created).~~
- ~~[ ] Required Standard Operating Procedure (SOP) is added.~~
- ~~[ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: pkg/client/aws/client_test.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+			},
+			want: &MockClientFactory{
+				mock: client,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			Expect(NewMockClientFactory(tt.args.client)).To(Equal(tt.want))
```
Comment (ziccardi) [inline pkg/client/aws/client_test.go:150]: see previous comment

File: pkg/client/aws/client_test.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := tt.args.client.ListHostedZonesByNameInput(testValue)
+			// only testing negative case here. The test should fail. This functionality is covered by integration tests
+			Expect(err != nil).To(Equal(tt.wantErr))
```
Comment (ziccardi) [inline pkg/client/aws/client_test.go:206]: See previous comment on same piece of code

File: pkg/client/aws/client_test.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := tt.args.client.ChangeResourceRecordSets(testValue, &route53.ChangeBatch{})
+			// only testing negative case here. The test should fail. This functionality is covered by integration tests
+			Expect(err != nil).To(Equal(tt.wantErr))
```
Comment (ziccardi) [inline pkg/client/aws/client_test.go:234]: See previous comment on same piece of code

**Review by valdar — APPROVED** [review]

File: pkg/client/aws/client_test.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+			},
+			wantErr: false,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := tt.fields.f.NewClient(tt.args.credentials, tt.args.region)
```
[the remark this answers is not in the document]
Comment (pawelpaszki) [reply pkg/client/aws/client_test.go]: you are right. I have removed that test

File: pkg/client/aws/client_test.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := tt.args.client.GetChange(testValue)
+			// only testing negative case here. The test should fail. This functionality is covered by integration tests
+			Expect(err != nil).To(Equal(tt.wantErr))
```
[the remark this answers is not in the document]
Comment (pawelpaszki) [reply pkg/client/aws/client_test.go]: this is changed now. Resolving this comment

**Review by ziccardi — APPROVED** [review]
lgtm

**Pull request merged**